### PR TITLE
Glutin update

### PR DESCRIPTION
**Pull request by paulrouget**
[Review on Reviewable](https://reviewable.io/reviews/servo/servo/7096)

### CODE REVIEW

File: ports/glutin/window.rs [hunk 1]
```diff
@@ -98,7 +98,10 @@ impl Window {
                             .with_parent(parent)
                             .build()
                             .unwrap();
-        unsafe { glutin_window.make_current() };
+        match unsafe { glutin_window.make_current() } {
```
Comment (glennw) [inline ports/glutin/window.rs]: This could be ...make_current().expect("Failed to make context current!");

File: ports/glutin/window.rs [hunk 2]
```diff
@@ -519,7 +522,7 @@ impl WindowMethods for Window {
     }
 
     fn present(&self) {
-        self.window.swap_buffers()
+        self.window.swap_buffers().is_ok();
```
Comment (glennw) [inline ports/glutin/window.rs]: I think just .ok() is generally preferred for this.

File: ports/glutin/window.rs (hunk 1 above)
Comment (paulrouget) [inline ports/glutin/window.rs]: I can't use `expect`:

```
/Users/paul/github/servo/ports/glutin/window.rs:102:47: 102:88 error: use of unstable library feature 'result_expect': newly introduced (see issue #27277)
/Users/paul/github/servo/ports/glutin/window.rs:102         unsafe { glutin_window.make_current().expect("Failed to make context current!") }
                                                                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/paul/github/servo/ports/glutin/window.rs:102:47: 102:88 help: add #![feature(result_expect)] to the crate attributes to enable
```

File: ports/glutin/window.rs (hunk 2 above)
Comment (nox) [inline ports/glutin/window.rs]: The intent of that code isn't obvious, shouldn't we instead do "let _ = …"?

File: ports/glutin/window.rs (hunk 1 above)
Comment (nox) [inline ports/glutin/window.rs]: Just enable the feature.

File: ports/glutin/window.rs (hunk 2 above)
Comment (nox) [inline ports/glutin/window.rs]:
```
13:50 <Ms2ger> I prefer .unwrap()
13:50 <nox> Ms2ger: Yeah me too but I wasn't sure if that call is supposed to fail silently or not.
```

**Comment by paulrouget** [comment]
Addressed Nox' comments.

**Comment by nox** [comment]
@bors-servo r=glennw r+

Thanks for your contribution.

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 2 unresolved discussions, all commit checks successful.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/servo/servo/7096)

**Comment by paulrouget** [comment]
This still requires an update of Cargo.toml, and it depends on https://github.com/servo/glutin/pull/44

**Comment by paulrouget** [comment]
The glutin servo branch has been updated.

I ran `cargo update -p glutin`, and ended up with this new `Cargo.lock`. Does it look right?

**Comment by nox** [comment]
@paulrouget You must use ./mach update-cargo, AFAIK.

**Comment by glennw** [comment]
@paulrouget Yep, if you run ./mach update-cargo -p glutin like @nox mentioned, this will update the cargo.lock files for all ports. I have only quickly scanned the changed cargo.lock file, but it looks about right - lots of new dependencies from the upstream glutin :)

**Comment by paulrouget** [comment]
Updated with `mach update-cargo -p glutin`.

**Pull request merged**